public/app/features/explore/QueryLibrary/QueryLibraryContext.tsx

```diff
@@ -19,17 +17,33 @@ export type QueryLibraryContextType = {
    * @param datasourceFilters Data source names that will be used for initial filter in the library.
    * @param queryActionButton Action button will be shown in the library next to the query and can implement context
    *   specific actions with the library, like running the query or updating some query in the current app.
+   * @param options.context Used for tracking. Should identify the context this is called from, like 'explore' or
+   *   'dashboard'.
    */
-  openDrawer: (datasourceFilters: string[], queryActionButton: QueryActionButton) => void;
+  openDrawer: (
+    datasourceFilters: string[],
+    queryActionButton: QueryActionButton,
+    options?: { context?: string }
+  ) => void;
   closeDrawer: () => void;
   isDrawerOpen: boolean;
 
   /**
    * Opens a modal for adding a query to the library.
-   * @param query
+   * @param query Query to be saved
+   * @param options.onSave Callback that will be called after the query is saved.
+   * @param options.context Used for tracking. Should identify the context this is called from, like 'explore' or
+   *   'dashboard'.
    */
-  openAddQueryModal: (query: DataQuery) => void;
+  openAddQueryModal: (query: DataQuery, options?: { onSave?: () => void; context?: string }) => void;
   closeAddQueryModal: () => void;
+
+  /**
+   * Returns a predefined small button that can be used to save a query to the library.
+   * @param query
+   */
+  renderSaveQueryButton: (query: DataQuery) => ReactNode;
```

Comment:
I added this so I can move the SaveQueryButton into enterprise. The main thing it does is show a `new` badge to grab attention so I sort of assume it may change in the future and we may not need it, but for now this seemed the easiest solution.